common/query.go

```diff
@@ -6,13 +6,102 @@ package common
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/ohler55/ojg/jp"
 )
 
 // QueryArgs is a map of key-value pairs of arguments for a query
 type QueryArgs map[string]interface{}
 
+func (a *QueryArgs) AddFromText(text string) error {
+	parts := strings.SplitN(text, "=", 2)
+	if len(parts) != 2 {
+		return fmt.Errorf("\"=\" not found in \"%s\"", text)
+	}
+
+	name, value := parts[0], parts[1]
+	(*a)[name] = value
+	return nil
+}
+
+// GetString returns the value for the specified name as a string, or an error
+// if the value could not be interpreted as such.
+func (a QueryArgs) GetString(name string) (string, error) {
+	argVal, ok := a[name]
+	if !ok {
+		return "", fmt.Errorf("missing argument '%s'", name)
+	}
+
+	var result string
+	switch v := argVal.(type) {
+	case string:
+		result = v
+	case []interface{}:
+		result, ok = v[0].(string)
+		if !ok {
+			return "", fmt.Errorf("unexpected type for '%s'; must be a string", name)
+		}
+	default:
+		return "", fmt.Errorf("unexpected type for '%s'; must be a string; found: %T", name, v)
+	}
+
+	return result, nil
+}
+
+// GetStringSlice returns the value for the specified name as a string slice,
+// or an error if the value could not be interpreted as such.
+func (a QueryArgs) GetStringSlice(name string) ([]string, error) {
```

Comment:
I'd have called this `GetAsStringArray` (I find the use of `Slice` in this context a bit confusing, but it may be me...)

Reply:
I'm being consistent with Viper here (https://github.com/spf13/viper/tree/v1.7.1#getting-values-from-viper), but can change if you'd prefer.

Reply:
cool, thanks for the pointer.